x-pack/solutions/security/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/upgrade_with_preview.md

```diff
@@ -0,0 +1,706 @@
+    - [User gets notified after a new rule versions is released](#user-gets-notified-after-a-new-rule-versions-is-released)
+
+## Useful information
+
+### Tickets
+
+- [Users can Customize Prebuilt Detection Rules](https://github.com/elastic/security-team/issues/1974) (internal)
+- [Users can Customize Prebuilt Detection Rules: Milestone 3](https://github.com/elastic/kibana/issues/174168)
+- [Tests for prebuilt rule upgrade workflow](https://github.com/elastic/kibana/issues/202078)
+
+### Terminology
+
+- **CTA**: "call to action", usually a button, a link, or a callout message with a button, etc, that invites the user to do some action.
+
+  - **CTA to upgrade the prebuilt rule**: the button to upgrade the prebuilt rule currently shown in the Rule Upgrade flyout.
+
+- **Non-customized field**: a prebuilt rule's field that has the original value from the originally installed prebuilt rule.
+
+- **Customized field**: a prebuilt rule's field that has a value that differs from the original field value of the originally installed prebuilt rule.
+
+- **Non-customized rule**: a prebuilt rule that doesn't have any customized fields.
+
+- **Customized rule**: a prebuilt rule that has one or more customized fields.
+
+### Assumptions
+
+- Below scenarios only apply to prebuilt detection rules.
+- A prebuilt rule is shown in the Rule Upgrade table when there's a newer version of this rule in the currently installed package with prebuilt rules.
+
+### Functional requirements
+
+- User should be able to upgrade prebuilt rules one-by-one with the ability to preview:
+  - what updates they would receive from Elastic in the latest rule version;
+  - what user customizations they would retain in the rule;
+  - are there any conflicts between the updates from Elastic and the user customizations;
+  - what they should pay attention to in case there are any conflicts.
+- A preview should be shown in the Rule Upgrade flyout.
+- The Rule Upgrade flyout should contain a few tabs:
+  - The "Updates" tab.
+  - The "Overview" tab.
+  - The "Investigation guide" tab.
+- On the "Updates" tab:
+  - We should show the updates from Elastic and the user-customized fields.
+  - We should show only those fields that are [customizable](./shared_assets/customizable_rule_fields.md).
+  - We shouldn't show technical fields and those that are [not customizable](./shared_assets/non_customizable_rule_fields.md).
+- User should be able to upgrade a prebuilt rule that has some updates to [non-customizable fields](./shared_assets/non_customizable_rule_fields.md) in the latest version.
+- Any other fields that are not involved in the rule upgrade workflow, such as `enabled` or `exceptions`, should stay unchanged after rule upgrade.
+
+## Scenarios
+
+### Rule upgrade field preview
+
+#### Preview non-customized field that has an upgrade (AAB)
+
+**Automation**: Jest functional test for each \<field\>.
```

Comment:
What's the difference between "Jest functional tests" and "Jest integration tests" mentioned below?

Reply:
These are the same. Ofc we should stick to the same wording. May we should use `FrontEnd integration tests`? Thanks for catching.